share/spack/gitlab/cloud_pipelines/stacks/aws-pcluster-icelake/spack.yaml

```diff
@@ -0,0 +1,58 @@
+spack:
+  view: false
+
+  definitions:
+  - compiler_specs:
+    - gcc
+    - gettext
+
+  - compiler_target:
+    - '%gcc@7.3.1 target=x86_64_v3'
+
+  - optimized_configs:
+    # - gromacs
+    - lammps
+    # - mpas-model
+    - openfoam
+    # - palace
+    # - py-devito
+    # - quantum-espresso
+    # - wrf
+
+  - optimized_libs:
+    - mpich
+    - openmpi
+
+  specs:
+  - matrix:
+    - - $compiler_specs
+    - - $compiler_target
+  - $optimized_configs
+  # - $optimized_libs
+
+  mirrors: { "mirror": "s3://spack-binaries/develop/aws-pcluster" }
+
+  ci:
+    pipeline-gen:
+    - build-job:
```

Comment:
It seems like this part could be moved into `share/spack/gitlab/cloud_pipelines/configs/pcluster/ci.yaml` and included or added to the generate jobs configs.

Reply:
Here https://github.com/spack/spack/pull/37627/files#diff-70f9e236c9b2eff97832cfe086f0ba52c7d7fad02d03bf0c04189bc8dd95b1f0R107

Reply:
Yes, I can do that.